### PR TITLE
Replace ntp with ntpsec

### PR DESCRIPTION
**Pull request by Mno-hime**
Replace `ntp` classic with `ntpsec`. Mostly it's patches & files from OmniOS.

### CODE REVIEW

File: components/network/ntpsec/files/ntpsec [hunk 1]
```diff
@@ -22,14 +22,16 @@
...
 #
 
-# Standard prolog
+# Standard prologue
 #
 . /lib/svc/share/smf_include.sh
 
 if [ -z $SMF_FMRI ]; then
-        echo "SMF framework variables are not initialized."
+        echo "SMF framework variables are not initialised."
```
Comment (pyhalov) [inline components/network/ntpsec/files/ntpsec]: Don't do it, it's English/American English mess.  :)

**Comment by pyhalov** [comment]
@xen0l How do you feel about it?

File: components/network/ntpsec/files/ntpsec (hunk 1 above)
Comment (Mno-hime) [reply components/network/ntpsec/files/ntpsec]: That came from the British section of OmniOS :).

**Comment by jimklimov** [comment]
Just in case: is this a drop-in replacement regarding upgrades (config file formats and locations, service names for people who scripted or depended something on old service)? If something is "no", maybe old implem should not be obsoleted.

Also, is this a scripted (python) or binary implem? How responsive (system under stress/load) and resource-hungry is it in comparison to old binary?

**Comment by Mno-hime** [comment]
> Just in case: is this a drop-in replacement regarding upgrades (config file formats and locations, service names for people who scripted or depended something on old service)? If something is "no", maybe old implem should not be obsoleted.

As I understand it `ntpsec` uses the same configuration format as `ntp` classic. To quote [FAQ](https://ntpsec.org/FAQ.html):

> _Will NTPsec fully interoperate with NTP Classic?_
> 
> NTPsec already fully interoperates with NTP Classic. They use the same protocols and the same time adjustment algorithms. User-visible differences are minor, and consist mostly of obsolete code being dropped and a few tools renamed to reduce NTPsec’s footprint in global namespace.

[Also](https://docs.ntpsec.org/latest/ntpsec.html):

> We retain, however, almost full compatibility and interoperation with NTP Classic. The qualification "almost" is required mainly because we do not support the Autokey (RFC 5906) public-key encryption scheme. It had interoperability and exploitable vulnerability issues too severe to be patched. We have also dropped broadcast and anycast modes because they cannot be secured.

> Also, is this a scripted (python) or binary implem?

```
/usr/bin/ntpdate: POSIX shell script, ASCII text executable
/usr/bin/ntpdig: Python script, ASCII text executable
/usr/bin/ntpfrob: ELF 64-bit LSB executable, x86-64, version 1 (Solaris), dynamically linked, interpreter /usr/lib/amd64/ld.so.1, stripped
/usr/bin/ntpkeygen: Python script, ASCII text executable
/usr/bin/ntplogtemp: Python script, ASCII text executable
/usr/bin/ntpmon: Python script, ASCII text executable
/usr/bin/ntpq: Python script, ASCII text executable
/usr/bin/ntpsnmpd: Python script, ASCII text executable
/usr/bin/ntpsweep: Python script, ASCII text executable
/usr/bin/ntptime: ELF 64-bit LSB executable, x86-64, version 1 (Solaris), dynamically linked, interpreter /usr/lib/amd64/ld.so.1, stripped
/usr/bin/ntptrace: Python script, ASCII text executable
/usr/bin/ntpwait: Python script, ASCII text executable
/usr/sbin/ntpd: ELF 64-bit LSB executable, x86-64, version 1 (Solaris), dynamically linked, interpreter /usr/lib/amd64/ld.so.1, stripped
```

> How responsive (system under stress/load)

My old laptop is development box, under considerable load at times and I have no problems running `ntpsec`.

> and resource-hungry is it in comparison to old binary?

Very low/minimal?

```
{global} newman@lenovo:~ $ svcs -vp ntp
STATE          NSTATE        STIME    CTID   FMRI
online         -             19:19:33    331 svc:/network/ntp:default
               19:19:33      194 ntpd
```

```
USER       PID %CPU %MEM   SZ  RSS TT       S    START  TIME COMMAND
root       194  0.0  0.1 8360 3480 ?        S 19:19:33  0:00 /usr/sbin/ntpd -p
```

```
{global} newman@lenovo:~ $ pfexec lsof -p 194
COMMAND PID USER   FD   TYPE             DEVICE SIZE/OFF       NODE NAME
ntpd    194 root  cwd   VDIR                          21       1463 /root
ntpd    194 root  txt   VREG                      549304     406581 /usr/sbin/ntpd
ntpd    194 root  txt   VREG                      501448     666564 /lib/amd64/ld.so.1
ntpd    194 root  txt   VREG                     2986856     364193 /lib/openssl/default/amd64/libcrypto.so.1.0.0
ntpd    194 root  txt   VREG                       81288     666579 /lib/amd64/libmd.so.1
ntpd    194 root  txt   VREG                      211128     666571 /lib/amd64/libdevinfo.so.1
ntpd    194 root  txt   VREG                      131256     666673 /lib/amd64/libnvpair.so.1
ntpd    194 root  txt   VREG                      143536     666897 /usr/lib/amd64/libsmbios.so.1
ntpd    194 root  txt   VREG                       45344     666576 /lib/amd64/libgen.so.1
ntpd    194 root  txt   VREG                       28424     666580 /lib/amd64/libmp.so.2
ntpd    194 root  txt   VREG                       60744     666758 /lib/amd64/libuutil.so.1
ntpd    194 root  txt   VREG                       21000     666558 /lib/amd64/libpthread.so.1
ntpd    194 root  txt   VREG                      550392     660817 /lib/amd64/libm.so.2
ntpd    194 root  txt   VREG                      288608     666082 /lib/amd64/libscf.so.1
ntpd    194 root  txt   VREG                      129600     666495 /lib/amd64/libsec.so.1
ntpd    194 root  txt   VREG              577,3     4096 3654020701 /var/run/tzsync
ntpd    194 root  txt   VREG                        5600     666554 /lib/amd64/libdl.so.1
ntpd    194 root  txt   VREG                      112704     666111 /usr/lib/amd64/libidmap.so.1
ntpd    194 root  txt   VREG                       18912     666487 /lib/amd64/libavl.so.1
ntpd    194 root  txt   VREG                      718184     666581 /lib/amd64/libnsl.so.1
ntpd    194 root  txt   VREG                       62472     666539 /usr/lib/amd64/libdns_sd.so.1
ntpd    194 root  txt   VREG                        8680     666559 /lib/amd64/librt.so.1
ntpd    194 root  txt   VREG                     2094312     666489 /lib/amd64/libc.so.1
ntpd    194 root  txt   VREG                       45696     336274 /usr/gcc/6/lib/amd64/libssp.so.0.0.0
ntpd    194 root  txt   VREG                      103912     666754 /lib/amd64/libsocket.so.1
ntpd    194 root    0r  VCHR              155,2            81264648 /devices/pseudo/mm@0:null
ntpd    194 root    1r  VCHR              155,2            81264648 /devices/pseudo/mm@0:null
ntpd    194 root    2r  VCHR              155,2            81264648 /devices/pseudo/mm@0:null
ntpd    194 root    3w  VCHR            149,118            78118916 /devices/pseudo/log@0:conslog->LOG
ntpd    194 root    4r  DOOR                         0t0         25 /var/run/name_service_door (door to nscd[3331]) (FA:->0xffffff06d5ff1080)
ntpd    194 root    5w  VREG                       20297      92344 /var/ntp/ntpstats/peerstats.20190504
ntpd    194 root    6w  VREG                        1620      92549 /var/ntp/ntpstats/loopstats.20190504
ntpd    194 root   20u  IPv6 0xffffff06df3e3a00      0t0        UDP *:ntp (Idle)
ntpd    194 root   21u  IPv4 0xffffff06d68ab5c0      0t0        UDP *:ntp (Idle)
ntpd    194 root   22u  IPv4 0xffffff06cb7465c0      0t0        UDP localhost:ntp (Idle)
ntpd    194 root   23u  IPv4 0xffffff075c911580      0t0        UDP 192.168.1.177:ntp (Idle)
ntpd    194 root   24u  IPv6 0xffffff06cbdd8a00      0t0        UDP lenovo:ntp (Idle)
ntpd    194 root   25u  IPv6 0xffffff06d67d3080      0t0        UDP [fe80::3e97:eff:fe0b:f35b]:ntp (Idle)
ntpd    194 root   26u  IPv6 0xffffff06f83e1a80   0t4620      ROUTE (Wait_Data_Xfr)
```

**Comment by Mno-hime** [comment]
Updated to ntpsec 1.1.4, release notes: https://blog.ntpsec.org/2019/06/21/version-1.1.4.html.

**Comment by pyhalov** [comment]
From what I read, RedHat is replacing ISC NTP with chrony.
What other news in this field do we have?

I don't want to make so radical move without a bit more discussion.

**Comment by pyhalov** [comment]
https://chrony.tuxfamily.org/comparison.html

**Pull request closed without merging**